README.md

```diff
@@ -95,35 +95,65 @@ These operations add to or update data in Immutable's databases and typically re
 Signers are abstractions of user accounts that can be used to sign transactions. A user's private key is required to generate them.
 
 There are two ways to get signers in your application:
-
 1. [Generate your own by obtaining and using the user's private keys](#1-generate-your-own-signers)
 2. [Use our Wallet SDK to connect to a user's wallet application](#2-generate-signers-using-the-wallet-sdk)
 
 The first option, where an application obtains a user's private key directly, is risky because these keys allow anyone in possession of them full control of an account.
 
 The second option provides an application with an interface to the user's account by prompting the user to connect with their wallet application (ie. mobile or browser wallet). Once connected the app can begin asking the user to sign transactions and messages without having to reveal their private key.
 
-As Immutable X enables applications to execute signed transactions on both Ethereum (layer 1) and StarkEx (layer 2), signers are required for both these layers.
+### Pre-requisite: User must have Ethereum (L1) and Stark (L2) keys
 
-### 1. Generate your own signers
+As Immutable X enables applications to execute signed transactions on both Ethereum (layer 1) and StarkEx (layer 2), signers are required for both these layers. In order to generate an Ethereum or Stark signer, a user's Ethereum or Stark private key is required.
+
+The Core SDK can be used to generate a Stark key for a user. Previously, this key was deterministically generated so that the same key could always be obtained using the user's Ethereum key. This meant that if someone had access to a user's L1 (Ethereum) private keys, they could also obtain the user's L2 (Stark) private keys. This was [changed in v1.0.0-beta3](https://docs.x.immutable.com/sdk-docs/core-sdk-ts/core-sdk-migration-guide/#what-has-changed-in-core-sdk-v100-beta3) of the Core SDK, so that Stark keys are now randomly generated and non-reproducible.
```

Comment:
The Core SDK can randomly generate a Stark key for a user.

When using the Link SDK on a website that has implemented the ImmutableX global order book, like a marketplace, the Stark key is determined using a seed. This enables users to sign in to any website that implements the global order book using Link and interact with the ImmutableX StarkEx Layer2. In other words, the Link SDK or other wallets didn't need to persist the Stark key since the previously generated key could always be obtained using the user's Ethereum key when they sign for a request. 

This meant that if someone had access to a user's L1 (Ethereum) private keys, they could also obtain their L2 (Stark) private keys. This was [changed in v1.0.0-beta3](https://docs.x.immutable.com/sdk-docs/core-sdk-ts/core-sdk-migration-guide/#what-has-changed-in-core-sdk-v100-beta3) of the Core SDK so that Stark keys are now randomly generated and non-reproducible. The trade-off with generating a random stark key is that the user is responsible for storing the key and using that to sign for L2 requests.